### DISCUSSION TITLE
Assertions on last rendered blade

### DISCUSSION THREAD

**Discussion by julesgraus** [livewire/livewire#5599] [3. Feature ideas]
I've notice that even tough I test as much as I possible can using Livewire::test, the component still not always works like expected when using it in the browser. Mainly because I did forget to add wire:model statements or use non existing public properties in them.

I would like to help to add some assertion methods to Livewire::test that help you assert these kind of things. Example

```
Livewire::test(CreatePost::class)
    ->set('title', 'foo')
    ->assertPropertyBound('title')
```

In this case, assertPropertyBound would assert that there is a wire:model="title" statement in the last rendered blade.
This is how I think I would be able to build this functionality: Inside the Livewire::Test code there is a ```lastRenderedView``` and ```lastRenderedDom``` property somewhere. I think I would be able to use something like symphony's dom crawler package or regex, to scan the HTML for the ```wire:model="title"``` statement. If I could find it, I could make the assertion.

When this is possible, we could also implement methods like ```assertActionBound()```

Is it okay if I would try to build this and create a pull request?

**Comment by joshhanley** [comment]
@julesgraus have a look at this pacakge, I believe it adds what you're after https://github.com/christophrumpel/missing-livewire-assertions

Hope this helps!

**Reply by julesgraus** [reply]
Since my pull request #5604 was closed because we've moved on to Livewire version 3, I am creating a new pull request after modifying the code to be compatible with version 3. I know there are packages for version 2 that provide the desired functionality. But I want SOME of that functionality to be part of livewire itself. For the reasons mentioned when starting this discussion.

**Reply by julesgraus** [reply]
Here you can find the pull request: https://github.com/livewire/livewire/pull/5858
